src/serializer/ResidualHeapVisitor.js

```diff
@@ -113,7 +113,7 @@ export class ResidualHeapVisitor {
     }
   }
 
-  visitObjectProperties(obj: ObjectValue): void {
+  visitObjectProperties(obj: ObjectValue, kind: ?ObjectKind): void {
```

Comment:
`?ObjectKind` allows for `null` and `undefined`, right?
Also, there are a number of callsites that omit the argument. I understand that Flow allows for this, but let's make it more explicit, and make the argument `kind?: ObjectKind` which also only allows `undefined` but not `null`.